### PR TITLE
v0.38 changelog

### PR DESCRIPTION
**Pull request by isaacdevlugt**

### CODE REVIEW

File: doc/releases/changelog-0.38.0.md [hunk 1]
```diff
@@ -6,41 +6,317 @@
...
+          # 2 + 1 = 3, or |11> 
+          return qml.sample(wires=[0, 1])
+      ```
+
+      ```pycon
+      >>> circuit()
+      tensor([1, 1], requires_grad=True)
+      ```
+
+    * `qml.PhaseAdder` is similar to `qml.Adder`, but it performs in-place modular addition in the Fourier
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.38.0.md:184]: Not married to having a code example for `qml.PhaseAdder`

File: doc/releases/changelog-0.38.0.md [hunk 2]
```diff
@@ -6,108 +6,461 @@
...
+  * **In-place operations**: :math:`\text{SomeOp}(k, mod) \vert x \rangle = \vert x * k \; \text{modulo} \; mod \rangle` 
+    where :math:`*` denotes addition (:math:`+`) or multiplication (:math:`\times`). Each of the following 
+    operations require that you specify `k` and `x_wires` (the wires we operate on and store the result).
+    Optionally, you can override the default modulus value (`2**len(x_wires)`) with the `mod` keyword 
+    argument.
 
-* The `qml.OutAdder` and `qml.ModExp` templates are added to perform out-of-place modular addition and modular exponentiation.
-  [(#6121)](https://github.com/PennyLaneAI/pennylane/pull/6121)
+    * `qml.Adder` performs in-place modular addition: 
+      :math:`\text{Adder}(k, mod)\vert x \rangle = \vert x + k \; \text{modulo} \; mod \rangle`.
```
Comment (josh146) [inline doc/releases/changelog-0.38.0.md:166]: similar comment here re: the repeated mod/modulo!

File: doc/releases/changelog-0.38.0.md (hunk 1 above)
Comment (josh146) [reply doc/releases/changelog-0.38.0.md:185]: I would say you can remove code examples from all. Just have a bullet point for each function description, and then have a single, general code example at the end.

File: doc/releases/changelog-0.38.0.md (hunk 2 above)
Comment (josh146) [reply doc/releases/changelog-0.38.0.md:166]: and below

**Review by josh146 — COMMENTED** [review]
Have reviewed until improvements :) Looks great, all my comments are minor!

File: doc/releases/changelog-0.38.0.md [hunk 3]
```diff
@@ -6,108 +6,423 @@
...
+<h4>Quantum arithmetic operations 🧮</h4>
+
+* Several new operator templates have been added to PennyLane that let you perform quantum arithmetic 
+  operations.
+
+  * **In-place operations**: :math:`\text{SomeOp}(k, mod) \vert x \rangle = \vert x * k \; \text{modulo} \; mod \rangle` 
+    where :math:`*` denotes addition (:math:`+`) or multiplication (:math:`\times`). Each of the following 
+    operations require that you specify `k` and `x_wires` (the wires we operate on and store the result).
+    Optionally, you can override the default modulus value (`2**len(x_wires)`) with the `mod` keyword 
+    argument.
```
[the remark this answers is not in the document]
Comment (josh146) [reply doc/releases/changelog-0.38.0.md:163]: Same! I also found it hard to follow. There is a bit too much text, maybe it can be made more succinct?

File: doc/releases/changelog-0.38.0.md (hunk 1 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-0.38.0.md:185]: Yeah, I'm in favour of that @josh146. Too verbose at the moment 😅

File: doc/releases/changelog-0.38.0.md [hunk 4]
```diff
@@ -4,112 +4,362 @@
...
+      ```
+
+    * difference with `-` or `difference()`:
+
+      ```python
+      >>> wires1.difference(wires2) # or wires1 - wires2
+      Wires([1])
+      ```
 
 <h4>Quantum arithmetic operations 🧮</h4>
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.38.0.md:113]: @josh146 @DSGuala thoughts on how this is restructured?

File: doc/releases/changelog-0.38.0.md [hunk 5]
```diff
@@ -4,112 +4,362 @@
...
+
+      return qml.sample(wires=wire_reg["output_wires"])
+  ```
+
+  ```
+  >>> circuit()
+  array([0, 1, 0])
+  ```
+
+<h4>Converting noise models from Qiskit ♻️</h4>
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.38.0.md:175]: Moved this below registers and arithmetic

File: doc/releases/changelog-0.38.0.md (hunk 1 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-0.38.0.md]: See new commits!

File: doc/releases/changelog-0.38.0.md [hunk 6]
```diff
@@ -4,112 +4,396 @@
...
+
+  ```python
+  dev = qml.device("default.qubit", shots=1)
+
+  wire_reg = qml.registers({
+      "x_wires": 2, # |x>: stores the result of 2 + 1 = 3
+      "y_wires": 2, # |y>: multiples x by 3
+      "output_wires": 3, # stores the result of (2 + 1) * 3 m 7 = 2
+      "work_wires": 2 # for qml.OutMultiplier
+  })
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.38.0.md:150]: Nice to use registers here! But I get if we don't want to use it in this setting because it might take away / be distracting from the arithmetic features

File: doc/releases/changelog-0.38.0.md (hunk 4 above)
Comment (DSGuala) [reply doc/releases/changelog-0.38.0.md:113]: Much easier to follow this way :+1:

File: doc/releases/changelog-0.38.0.md (hunk 6 above)
Comment (DSGuala) [reply doc/releases/changelog-0.38.0.md:150]: As a reader, it took me a while to parse the use of registers for different parts of `(2 + 1) * 3 mod 7 = 2`. But I think it would be even harder with lists of wires :thinking:

**Review by DSGuala — APPROVED** [review]

File: doc/releases/changelog-0.38.0.md [hunk 7]
```diff
@@ -4,112 +4,396 @@
...
+  def circuit(lcu, control, alphas, unitaries):
+      qml.PrepSelPrep(lcu, control)
+      qml.adjoint(prep_sel_prep)(alphas, unitaries)
+      return qml.state()
+  ```
+
+  ```pycon
+  >>> import numpy as np
+  >>> np.round(circuit(lcu, control, alphas, unitaries), decimals=2)
+  [ 1.+0.j -0.+0.j -0.+0.j -0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j]
```
Comment (justinpickering) [inline doc/releases/changelog-0.38.0.md]: I get what I believe is a PL tensor `tensor([ 1.+0.j, -0.+0.j,  0.+0.j, -0.+0.j,  0.+0.j,  0.+0.j,  0.+0.j,
         0.+0.j], requires_grad=True)`. Not sure if this should be updated, or if we default to a state vector for examples.

**Review by justinpickering — APPROVED** [review]
Looks great :+1: just two small comments below!

File: doc/releases/changelog-0.38.0.md [hunk 8]
```diff
@@ -4,112 +4,396 @@
...
+
+  Under the hood, PennyLane converts each quantum error in the Qiskit noise model into an equivalent 
+  `qml.QubitChannel` operator with the same canonical 
+  [Kraus representation](https://en.wikipedia.org/wiki/Quantum_operation#Kraus_operators). Currently, 
+  noise models in PennyLane do not support readout errors. As such, those will be skipped during conversion
+  if they are present in the Qiskit noise model.
+
+  Make sure to `pip install pennylane-qiskit` to access this new feature!
+
+<h4>Substantial upgrades to MCMs using tree-traversal 🌳</h4>
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.38.0.md:216]:
```suggestion
<h4>Substantial upgrades to mid-circuit measurements using tree-traversal 🌳</h4>
```

File: doc/releases/changelog-0.38.0.md [hunk 9]
```diff
@@ -4,112 +4,396 @@
...
+
+* The `"tree-traversal"` algorithm for MCMs on `default.qubit` has been internally redesigned for better
```
Comment (isaacdevlugt) [inline doc/releases/changelog-0.38.0.md:218]:
```suggestion
* The `"tree-traversal"` algorithm for mid-circuit measurements (MCMs) on `default.qubit` has been internally redesigned for better 
```

File: doc/releases/changelog-0.38.0.md [hunk 10]
```diff
@@ -4,112 +4,396 @@
...
-  [(#6108)](https://github.com/PennyLaneAI/pennylane/pull/6108/)
+<h4>Creating spin Hamiltonians</h4>
+
+* Three new functions are now available for creating commonly-used spin Hamiltonians in PennyLane:
+  [(#6106)](https://github.com/PennyLaneAI/pennylane/pull/6106)
+  [(#6128)](https://github.com/PennyLaneAI/pennylane/pull/6128)
+
+  * `qml.spin.transverse_ising`: creates the [transverse-field Ising model](https://en.wikipedia.org/wiki/Transverse-field_Ising_model) Hamiltonian.
+  * `qml.spin.heisenberg`: creates the [Heisenberg model](https://en.wikipedia.org/wiki/Quantum_Heisenberg_model) Hamiltonian.
+  * `qml.spin.fermi_hubbard`: creates the [Fermi-Hubbard model](https://en.wikipedia.org/wiki/Hubbard_model) Hamiltonian.
```
Comment (josh146) [inline doc/releases/changelog-0.38.0.md:270]: This section is not rendering correctly, and I can't tell why 🤔 

image

File: doc/releases/changelog-0.38.0.md [hunk 11]
```diff
@@ -4,112 +4,396 @@
...
-* The `qml.PrepSelPrep` template is added. The template implements a block-encoding of a linear
-  combination of unitaries.
+* A new template called `qml.PrepSelPrep` has been added that implements a block-encoding of a linear
+  combination of unitaries. 
   [(#5756)](https://github.com/PennyLaneAI/pennylane/pull/5756)
   [(#5987)](https://github.com/PennyLaneAI/pennylane/pull/5987)
 
+  This operator acts as a nice wrapper for having to perform `qml.StatePrep`, `qml.Select`, and `qml.adjoint(qml.StatePrep)`
+  in succession, which is quite common in many quantum algorithms. Here is an example showing the equivalence
+  between using `qml.PrepSelPrep` and `qml.StatePrep`, `qml.Select`, and `qml.adjoint(qml.StatePrep)`.
```
Comment (josh146) [inline doc/releases/changelog-0.38.0.md:317]: @isaacdevlugt worth linking the LCU demo? https://pennylane.ai/qml/demos/tutorial_lcu_blockencoding/

File: doc/releases/changelog-0.38.0.md [hunk 12]
```diff
@@ -4,112 +4,396 @@
...
+  >>> np.round(circuit(lcu, control, alphas, unitaries), decimals=2)
+  tensor([1.+0.j -0.+0.j -0.+0.j -0.+0.j  0.+0.j  0.+0.j  0.+0.j  0.+0.j], requires_grad=True)
+  ```
+
 <h4>QChem improvements</h4>
 
 * Molecules and Hamiltonians can now be constructed for all the elements present in the periodic table.
   [(#5821)](https://github.com/PennyLaneAI/pennylane/pull/5821)
 
+  This new feature is made possible by integrating with the [`basis-set-exchange` package](https://pypi.org/project/basis-set-exchange/).
```
Comment (josh146) [inline doc/releases/changelog-0.38.0.md:350]:
```suggestion
  This new feature is made possible by integrating with the [basis-set-exchange package](https://pypi.org/project/basis-set-exchange/).
```

**Review by josh146 — APPROVED** [review]
Just two minor rendering issues!

File: doc/releases/changelog-0.38.0.md (hunk 11 above)
Comment (isaacdevlugt) [inline doc/releases/changelog-0.38.0.md:317]:
```suggestion
  in succession, which is quite common in many quantum algorithms (e.g., [LCU and block encoding](https://pennylane.ai/qml/demos/tutorial_lcu_blockencoding/)). Here is an example showing the equivalence
  between using `qml.PrepSelPrep` and `qml.StatePrep`, `qml.Select`, and `qml.adjoint(qml.StatePrep)`.
```

File: doc/releases/changelog-0.38.0.md (hunk 10 above)
Comment (isaacdevlugt) [reply doc/releases/changelog-0.38.0.md]: Screenshot 2024-08-30 at 5 49 59 PM
resolved! 🤷
